modules/reflection/src/QC_Class.qpp

```diff
@@ -116,6 +134,177 @@ string Class::getPathName() [flags=RET_VALUE_ONLY] {
+    TempEncodingHelper nname(name, QCS_DEFAULT, xsink);
+    if (*xsink) {
+        return QoreValue();
+    }
+
+    // get read access to program object
+    QoreExternalProgramContextHelper tch(xsink, c->pgm);
+    if (*xsink) {
+        return QoreValue();
+    }
+
+    QoreReflectionHierarchyMethod* method;
+    ReferenceHolder<QoreReflectionMethod> m(method = new QoreReflectionHierarchyMethod(c->cls, nname->c_str(), xsink,
+        MT_Normal), xsink);
+    if (*xsink) {
+        return QoreValue();
+    }
+
+    ReferenceHolder<QoreHashNode> rv(new QoreHashNode(hashdeclMethodAccessInfo, xsink), xsink);
+    rv->setKeyValue("method", get_method_object(m, xsink), xsink);
+    rv->setKeyValue("access", get_access(method->getClassAccess()), xsink);
+    rv->setKeyValue("access_string", new QoreStringNode(get_access_string(method->getClassAccess())), xsink);
+
+    return rv.release();
+}
+
+//! Returns information about the given static method if it's accessible in the class hierarchy or throws an exception if the method does not exist
+/** @par Example:
+    @code{.py}
+MethodAccessInfo info = cls.getStaticMethod(name);
+printf("find('%s::%s()': found '%s::%s()'; class access %y, variants: %d\n", cls.getName(), name,
+    info.method.getClass().getName(), info.access_string, info.method.getVariants().size());
+    @endcode
+
+    Searches the class hierarchy for static methods for the given method name
+
+    @param name the name of the method
+
+    @return information about the given method; the access information in the hash is the access of the class
+    containing the method; to get the access modifier of any method variant, the maximum of the access value provided
+    here and the access value of the variant must be taken.
+
+    @throw PROGRAM-ERROR thrown if the @ref Qore::Program "Program" object holding the class has been destroyed
+    @throw METHOD-ERROR unknown method
+    @throw ENCODING-CONVERSION-ERROR this exception could be thrown if the string argument is not in the
+    @ref default_encoding "default character encoding" and an error occurs during encoding conversion
+
+    @note Methods group variants with the same name; see
+    @ref Qore::Reflection::AbstractMethod::getVariant() "AbstractMethod::getVariant()" for a method to retrieve a
+    specific variant from a method.
+
+    @see
+    - findMethod()
+    - findNormalMethod()
+    - @ref AbstractClass::getMethod()
+*/
+hash<MethodAccessInfo> Class::findStaticMethod(string name) [flags=RET_VALUE_ONLY] {
+    TempEncodingHelper nname(name, QCS_DEFAULT, xsink);
+    if (*xsink) {
+        return QoreValue();
+    }
+
+    // get read access to program object
+    QoreExternalProgramContextHelper tch(xsink, c->pgm);
+    if (*xsink) {
+        return QoreValue();
+    }
+
+    QoreReflectionHierarchyMethod* method;
+    ReferenceHolder<QoreReflectionMethod> m(method = new QoreReflectionHierarchyMethod(c->cls, nname->c_str(), xsink,
```

Comment:
because `get_method_object()` takes a `ReferenceHolder<QoreReflectionMethod>&` arg and `ReferenceHolder<QoreReflectionMethod>&` and `ReferenceHolder<QoreReflectionHierarchyMethod>&` are not compatible (even though the template arguments are - I don't know how else to solve that, so I did it like this.

```
QoreObject* get_method_object(ReferenceHolder<QoreReflectionMethod>& m, ExceptionSink* xsink) {
    // ...
}
```

Reply:
well the solution would be maybe introduce a function, that would take one referenceHolder and creates a different referenceHolder casting the inner pointer (and incrementing a reference count).
Something like this https://en.cppreference.com/w/cpp/memory/shared_ptr/pointer_cast for shared_ptrs

So we would have 2 referenceHolders for the same pointer with different template types, reference count would be 2, .... 

but anyway, not needed now I guess